Source/WebCore/rendering/RenderLayerScrollableArea.cpp

```diff
@@ -2048,5 +2050,17 @@ void RenderLayerScrollableArea::invalidateScrollAnchoringElement()
         m_scrollAnchoringController->invalidateAnchorElement();
 }
 
+void RenderLayerScrollableArea::registerScrollbars()
+{
+    createScrollbarsController();
+    if (auto* horizontalScrollbar = this->horizontalScrollbar()) {
+        horizontalScrollbar->theme().registerScrollbar(*horizontalScrollbar);
```

Comment:
ScrollBar's constructor calls `theme().registerScrollbar(*this);` so why do we do this again here?

Reply:
Same thing as above we need to re-register the scrollbars when we change the ScrollbarsController type.

Reply:
I think what's confusing is that the roles of `ScrollbarTheme` and `ScrollbarsController` are not really clear. Scrollbars register themselves with the theme at creation; registering them again with (possibly a different) theme here seems error prone. Maybe `ScrollbarsController` should be responsible for registering a scrollbar with the theme?

Also, if the `ScollbarsController` can change, I'd prefer to see an explicit "null out the old one, make a new one" code pattern.